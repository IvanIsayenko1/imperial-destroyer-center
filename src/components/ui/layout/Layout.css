.layout {
  display: flex;
  flex-direction: column;
  min-height: calc(100vh - 24px);
  gap: 12px;
}

.layout main {
  flex: 1;
  min-width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.layout footer {
  margin-top: auto;
}
