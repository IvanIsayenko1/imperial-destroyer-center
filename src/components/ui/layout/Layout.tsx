import React from "react";
import { Outlet } from "react-router-dom";
import { Header } from "../header/Header";
import "./Layout.css";
import { Footer } from "../footer/Footer";

export const Layout: React.FC = () => {
  return (
    <div className="layout">
      <Header />
      <main>
        <Outlet />
      </main>
      <Footer />
    </div>
  );
};
