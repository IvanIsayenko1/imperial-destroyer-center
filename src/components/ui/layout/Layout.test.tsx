import { render, screen } from "@testing-library/react";
import { Layout } from "./Layout";
import { BrowserRouter } from "react-router-dom";

vi.mock("../header/Header", () => ({
  Header: () => <div>Header</div>,
}));

vi.mock("../footer/Footer", () => ({
  Footer: () => <div>Footer</div>,
}));

describe("Layout component", () => {
  it("should render Header", () => {
    render(
      <BrowserRouter>
        <Layout />
      </BrowserRouter>
    );
    expect(screen.getByText("Header")).toBeInTheDocument();
  });

  it("should render Footer", () => {
    render(
      <BrowserRouter>
        <Layout />
      </BrowserRouter>
    );
    expect(screen.getByText("Footer")).toBeInTheDocument();
  });

  it("should render Outlet for nested routes", () => {
    render(
      <BrowserRouter>
        <Layout />
      </BrowserRouter>
    );
    const outlet = screen.getByRole("main"); // The <main> tag contains the Outlet
    expect(outlet).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Layout />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
