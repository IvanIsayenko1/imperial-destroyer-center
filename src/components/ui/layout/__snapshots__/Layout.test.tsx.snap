// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <div>
      Header
    </div>
    <main />
    <div>
      Footer
    </div>
  </div>
</DocumentFragment>
`;
